### ISSUE TITLE
Cannot add ensures clauses to logic functions in stdlib

### ISSUE THREAD

**Issue by jhjourdan** [xldenis/creusot#367]
I'd like to add the following to `creusot-contracts`:

```
impl<T, A: Allocator> Model for Vec<T, A> {
    type ModelTy = Seq<T>;

    #[logic]
    #[trusted]
    #[ensures(result.len() <= @isize::MAX)]
    fn model(self) -> Self::ModelTy {
        pearlite! { absurd }
    }
}
```

(Note the `ensures` clause.)

However, this generates a proof obligation in dependent crates.

There are two issues here:
- There should be a way to add trusted ensures clauses (even on untrusted logical definitions).
- VC of crates on which we depend should not give rise to VCs.

**Comment by xldenis** [comment]
> However, this generates a proof obligation in dependent crates.

I'm not able to recreate this, I used this diff:

```
 diff --git a/creusot-contracts/src/std/vec.rs b/creusot-contracts/src/std/vec.rs
 index 44a7a2d..143b6fb 100644
 --- a/creusot-contracts/src/std/vec.rs
 +++ b/creusot-contracts/src/std/vec.rs
 @@ -14,6 +14,7 @@ impl<T, A: Allocator> Model for Vec<T, A> {
      type ModelTy = Seq<T>;
      #[logic]
      #[trusted]
 +    #[ensures(result.len() <= @usize::MAX)]
      fn model(self) -> Self::ModelTy {
          absurd
      }
```

**Comment by xldenis** [comment]
> Indeed, at the moment, I believe we can generate VCs for _untrusted_ logical functions in dependencies.

Apart from the refinement obligations I actually think this isn't the case.

**Comment by jhjourdan** [comment]
> I'm not able to recreate this, I used this diff:
Indeed, sorry.

**Comment by xldenis** [comment]
This is expected, `builtins` will literally substitute calls with the provided function. I would recommend using the `id` wrapper we had before instead.

**Comment by xldenis** [comment]
I will add an error message checking that you aren't using both a contract and `builtins` at the same time though.

**Comment by xldenis** [comment]
Closing as won't fix, but with an explicit error added in #374
